paasta_tools/cli/cmds/status.py

```diff
@@ -2407,9 +2407,13 @@ def paasta_status(args) -> int:
 
     with concurrent.futures.ThreadPoolExecutor(max_workers=20) as executor:
         tasks = [executor.submit(t[0], **t[1]) for t in tasks]  # type: ignore
-        for future in concurrent.futures.as_completed(tasks):  # type: ignore
-            return_code, output = future.result()
-            return_codes.append(return_code)
+        try:
+            for future in concurrent.futures.as_completed(tasks):  # type: ignore
+                return_code, output = future.result()
+                return_codes.append(return_code)
+        except KeyboardInterrupt:
+            executor._threads.clear()
+            concurrent.futures.thread._threads_queues.clear()
```

Comment:
unfortunate that we need to dig into internals here :/

we'll probably need to `# type: ignore` here + add a comment about why we had to resort to this

Reply:
There might be a cleaner way of doing this. But no clue....

Reply:
I think this way is fine (i couldn't find anything better in my quick search) - but we should comment that we're reaching into python internals here 'cause there's no exposed interface to do this otherwise